mava/components/jax/training/step.py

```diff
@@ -140,8 +199,9 @@ def on_training_init_start(self, trainer: SystemTrainer) -> None:
             None.
         """
         # Note (dries): Assuming the batch and sequence dimensions are flattened.
```

Comment:
or are you of the opinion that the order of callbacks handles this?

Reply:
There's no functional issue with the circular dependency, but it just means the components are more interleaved than perhaps they should be: i.e. it might be poor design